src/main/kotlin/no/nb/mlt/wls/infrastructure/OutboxProcessor.kt

```diff
@@ -0,0 +1,134 @@
+
+        val processedEvent = outboxRepository.markAsProcessed(event)
+        logger.info { "Marked event as processed: $processedEvent" }
+    }
+
+    private fun handleOrderDeleted(event: OrderDeleted) {
+        TODO("Not yet implemented")
+    }
+
+    private suspend fun handleOrderUpdated(event: OrderUpdated) {
+        logger.info { "Processing OrderUpdated: $event" }
+        val updatedOrder = event.updatedOrder
+        val items =
+            itemRepository.getItems(
+                updatedOrder.orderLine.map { it.hostId },
+                updatedOrder.hostName
+            )
+
+        mapItemsOnLocation(items).forEach { (storageSystemFacade, itemList) ->
+            if (storageSystemFacade == null) {
+                logger.info { "Could not find a storage system to handle items: $itemList" }
+            }
+            storageSystemFacade?.updateOrder(updatedOrder)
+            // TODO - How to handle emails when an order is updated?
+            // createAndSendEmails(order)
+        }
+    }
+
+    private suspend fun handleItemCreated(event: ItemCreated) {
+        logger.info { "Processing ItemCreated: $event" }
+        val item = event.createdItem
+        val storageCandidates = mapCanStorageHandle(item)
+        if (storageCandidates.isEmpty()) {
+            logger.info { "Could not find a storage system to handle item: $item" }
+            return
+        }
+        storageCandidates.forEach {
+            it.createItem(item)
+        }
+    }
+
+    private suspend fun handleOrderCreated(event: OrderCreated) {
+        logger.info { "Processing OrderCreated: $event" }
+        val createdOrder = event.createdOrder
+
+        val items =
+            itemRepository.getItems(
+                createdOrder.orderLine.map { it.hostId },
+                createdOrder.hostName
+            )
+
+        mapItemsOnLocation(items).forEach { (storageSystemFacade, itemList) ->
+            if (storageSystemFacade == null) {
+                logger.info { "Could not find a storage system to handle items: $itemList" }
+            }
+
+            val orderCopy =
+                createdOrder.copy(
+                    orderLine =
+                        itemList.map {
+                            Order.OrderItem(it.hostId, Order.OrderItem.Status.NOT_STARTED)
+                        }
+                )
+            storageSystemFacade?.createOrder(orderCopy)
+            createAndSendEmails(orderCopy)
+        }
+    }
+
+    private suspend fun mapItemsOnLocation(items: List<Item>): Map<StorageSystemFacade?, List<Item>> {
+        return items.groupBy { item ->
+            storageSystems.firstOrNull { it.canHandleLocation(item.location) }
+        }
+    }
+
+    private suspend fun mapCanStorageHandle(item: Item): List<StorageSystemFacade> {
+        return storageSystems.mapNotNull {
+            if (it.canHandleLocation(item.location)) {
+                return@mapNotNull it
+            }
+            null
+        }
```

Comment:
Er det ikke like greit å bruke filter her?

```suggestion
        return storageSystems.filter { it.canHandleLocation(item.location) }
```